eng/DefaultGenApiDocIds.txt

```diff
@@ -7,9 +7,11 @@ T:System.ComponentModel.EditorAttribute
 T:System.ComponentModel.ToolboxItemAttribute
 T:System.ComponentModel.TypeDescriptionProviderAttribute
 T:System.Configuration.ConfigurationPropertyAttribute
+T:System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute
```

Comment:
If the purpose of this file is `These attributes should be excluded from reference assemblies.`, then `DynamicallyAccessedMembersAttribute` and `RequiresUnreferencedCodeAttribute` shouldn't be added to this list. We need them in ref assemblies.

Reply:
I thought the linker just depended on these being in the implementation. Thanks for the heads up, then I'll remove them and add them to the ref assemblies that need it.

Reply:
Linker does, but we have plans for a Roslyn analyzer that would consume the refs, so we've been trying to keep the refs annotated as well. Seems like some annotations slipped through.

Reply:
I see. So do we want to remove all the linker related attributes in sync, or just the ones I'm adding to this file as part of this PR?

Reply:
I'll fix the ones that I'm seeing on this PR by reverting this change and the other ones with the tooling updated from Arcade we can do in a follow up PR.

Reply:
Alright, I removed these 2 additions and fixed the missmatches.

Reply:
> So do we want to remove all the linker related attributes in sync, or just the ones I'm adding to this file as part of this PR?

Just the ones you're adding.

Which ones were you thinking of removing? DynamicDependencyAttribute and UnconditionalSuppressMessageAttribute don't have to be propagated to refs - they're implementation details that are relevant when analyzing the method bodies only and not interesting for dataflow analysis. Dataflow analysis only cares about DynamicallyAccessedMembers and RequiresUnreferencedCode.

> Alright, I removed these 2 additions and fixed the mismatches.

thank you!

Reply:
Great! Thanks for elaborating!